Sources/SentryCrash/Recording/Tools/SentryCrashLogger.h

```diff
@@ -129,115 +129,139 @@
  * ===============
  *
  * The C logger changes its behavior depending on the value of the preprocessor
- * define SentryCrashLogger_CBufferSize.
+ * define @c SentryCrashLogger_CBufferSize.
  *
- * If SentryCrashLogger_CBufferSize is > 0, the C logger will behave in an
+ * If @c SentryCrashLogger_CBufferSize is > 0, the C logger will behave in an
  * async-safe manner, calling write() instead of printf(). Any log messages that
  * exceed the length specified by SentryCrashLogger_CBufferSize will be
  * truncated.
  *
- * If SentryCrashLogger_CBufferSize == 0, the C logger will use printf(), and
+ * If @c SentryCrashLogger_CBufferSize == 0, the C logger will use printf(), and
  * there will be no limit on the log message length.
  *
- * SentryCrashLogger_CBufferSize can only be set as a preprocessor define, and
+ * @c SentryCrashLogger_CBufferSize can only be set as a preprocessor define, and
  * will default to 1024 if not specified during compilation.
  */
 
+/** The buffer size to use when writing log entries.
+ *
+ * If this value is > 0, any log entries that expand beyond this length will
+ * be truncated.
+ * If this value = 0, the logging system will dynamically allocate memory
+ * and never truncate. However, the log functions won't be async-safe.
+ *
+ * Unless you're logging from within signal handlers, it's safe to set it to 0.
+ */
+#ifndef SentryCrashLogger_CBufferSize
+#define SentryCrashLogger_CBufferSize 1024
+#endif
+
 // ============================================================================
 #pragma mark - (internal) -
 // ============================================================================
 
 #ifndef HDR_SentryCrashLogger_h
-#    define HDR_SentryCrashLogger_h
+#define HDR_SentryCrashLogger_h
 
-#    ifdef __cplusplus
+#ifdef __cplusplus
 extern "C" {
-#    endif
+#endif
 
-#    include <stdbool.h>
+#include <stdbool.h>
```

Comment:
m: We had some problems before by declaring headers after `extern "C"`. Do you think we can move this up?